lib/jsdom/virtual-console.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+
+var EventEmitter = require("events").EventEmitter;
+
+var vConsole = {};
+
+var emitters = {};
+var windows = [];
+
+// Returns an EventEmitter that emits on any console events
+vConsole.getVirtualConsole = function(window) {
+  var virtualConsole = new EventEmitter();
+
+  saveEmitterForWindow(virtualConsole, window);
+
+  virtualConsole.sendTo = function(anyConsole) {
```

Comment:
This should be a prototype method on a `VirtualConsole` class (that derives from `EventEmitter`) instead of having it be re-created every time.